elements/pf-popover/pf-popover.ts

```diff
@@ -0,0 +1,389 @@
+  @query('#trigger') private _slottedTrigger?: HTMLElement | null;
+  @query('#arrow') private _arrow?: HTMLElement | null;
+
+  #referenceTrigger?: HTMLElement | null = null;
+
+  #float = new FloatingDOMController(this, {
+    content: () => this._popover,
+    arrow: () => this._arrow,
+    invoker: () => this.#referenceTrigger || this._slottedTrigger
+  });
+
+  #slots = new SlotController(this, { slots: [null, 'icon', 'heading', 'body', 'footer'] });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('keydown', this._onKeydown);
+    document.addEventListener('click', this._outsideClick);
+  }
+
+  render() {
+    const { alignment, anchor, styles } = this.#float;
+    return html`
+      <div
+        id="container"
+        style="${styleMap(styles)}"
+        class="${classMap({
+          [anchor]: !!anchor,
+          [alignment]: !!alignment,
+        })}"
+      >
+        <slot id="trigger" @keydown=${this._onKeydown} @click=${this.show}></slot>
+        <dialog
+          id="popover"
+          aria-labelledby="heading"
+          aria-describedby="body"
+          aria-label=${ifDefined(this.label)}
+        > 
+          <div id="arrow"></div>
+          <div id="content" part="content">
+            ${this._renderCloseButton()} ${this._renderHeader()} ${this._renderBody()} ${this._renderFooter()}
+          </div>
+      </dialog>
+      </div>
+    `;
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    this.removeEventListener('keydown', this._onKeydown);
+    document.removeEventListener('click', this._outsideClick);
+
+    this.#referenceTrigger?.removeEventListener('click', this.show);
+    this.#referenceTrigger?.removeEventListener('keydown', this._onKeydown);
+  }
+
+  @bound private _onKeydown(event: KeyboardEvent) {
+    switch (event.key) {
+      case 'Escape':
+      case 'Esc':
+        event.preventDefault();
+        this.hide();
+        return;
+      case 'Enter':
+        if (event.target === this.#referenceTrigger || event.target === this._slottedTrigger) {
+          event.preventDefault();
+          this.show();
+        }
+        return;
+    }
+  }
+
+  @bound private _outsideClick(event: MouseEvent) {
+    const path = event.composedPath();
+    if (!path.includes(this) && !path.includes(this.#referenceTrigger as HTMLElement)) {
+      this.hide();
+    }
+  }
+
+  protected _triggerChanged(oldValue?: string, newValue?: string) {
+    if (oldValue) {
+      this.#referenceTrigger?.removeEventListener('click', this.show);
+      this.#referenceTrigger?.removeEventListener('keydown', this._onKeydown);
+    }
+    if (newValue) {
+      this.#referenceTrigger = (this.getRootNode() as Document | ShadowRoot).getElementById(newValue);
+      this.#referenceTrigger?.addEventListener('click', this.show);
+      this.#referenceTrigger?.addEventListener('keydown', this._onKeydown);
+    }
+  }
+
+  protected _renderCloseButton() {
```

Comment:
:thinking: can we inline these? we've tended to avoid breaking up the templates into private render methods, unless we had a compelling reason